plugins/cindy3d/src/js/Triangles.js

```diff
@@ -173,8 +185,22 @@ Triangles.prototype.render = function(viewer) {
     gl.texParameteri(gl.TEXTURE_2D, gl.TEXTURE_MIN_FILTER,
                      gl.LINEAR_MIPMAP_LINEAR);
     gl.hint(gl.GENERATE_MIPMAP_HINT, gl.NICEST);
-    gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, gl.RGBA, gl.UNSIGNED_BYTE,
-                  this.texture.img);
+    let /** @type {HTMLImageElement|HTMLCanvasElement|HTMLVideoElement} */ img
+        = this.texture.img;
+    if (w !== this.texture.width || h !== this.texture.height) {
```

Comment:
Probably it makes sense to have a
`if (this.texture.live || (this.img.generation < this.texture.generation)) {
this.img.generation = this.texture.generation;
...
`  here to avoid multiple rescalings of the same image.